metrics-exporter-dogstatsd/src/writer.rs

```diff
@@ -121,21 +123,25 @@ impl PayloadWriter {
         true
     }
 
-    fn write_trailing(&mut self, key: &Key, timestamp: Option<u64>) {
-        write_metric_trailer(key, timestamp, &mut self.buf, None);
+    fn write_trailing(&mut self, key: &Key, timestamp: Option<u64>, labels: Arc<Vec<Label>>) {
```

Comment:
Change `Arc<Vec<Label>>` to `&[Label]` just to be a little more generic, and probably call this `global_labels` just to be a little clearer that it's not the primary set of labels being written out.

Ditto this for every instance/method in the remainder of the file where take in the global labels.